receiver/awscontainerinsightreceiver/receiver.go

```diff
@@ -78,10 +78,11 @@ func (acir *awsContainerInsightReceiver) Start(ctx context.Context, host compone
 	}
 
 	if acir.config.ContainerOrchestrator == ci.EKS {
-		k8sDecorator, err := stores.NewK8sDecorator(ctx, acir.config.TagService, acir.config.PrefFullPodName, acir.config.AddFullPodNameMetricLabel, acir.config.AddContainerNameMetricLabel, acir.config.EnableControlPlaneMetrics, acir.settings.Logger)
-		acir.decorators = append(acir.decorators, k8sDecorator)
+		k8sDecorator, err := stores.NewK8sDecorator(ctx, acir.config.TagService, acir.config.PrefFullPodName, acir.config.AddFullPodNameMetricLabel, acir.config.AddContainerNameMetricLabel, acir.config.EnableControlPlaneMetrics, acir.config.KubeConfigPath, acir.config.HostIP, acir.config.HostName, acir.settings.Logger)
 		if err != nil {
-			return err
+			acir.settings.Logger.Warn("Unable to start K8s decorator", zap.Error(err))
```

Comment:
so failing to create the decorator won't be a hard failure?

Reply:
yeah its fail fast. container insights failing (due to kubelet/api-server) should not affect the other functionalities of the agent